diff --git a/lib/vmd/vmd.c b/lib/vmd/vmd.c
index c8a0c7906..1a6d3df3b 100644
--- a/lib/vmd/vmd.c
+++ b/lib/vmd/vmd.c
@@ -1127,6 +1127,8 @@ vmd_reset_root_ports(struct vmd_pci_bus *bus)
 
 		header = (volatile void *)(bus->vmd->cfg_vaddr + CONFIG_OFFSET_ADDR(bus->bus_number,
 					   devfn, 0, 0));
+		SPDK_ERRLOG("cfg_vaddr=%p bus_number=%d devfn=%d header=%p\n",
+			    bus->vmd->cfg_vaddr, bus->bus_number, devfn, header);
 		if (vmd_device_is_root_port(header) && !vmd_device_is_enumerated(header)) {
 			vmd_reset_base_limit_registers(header);
 		}
